app/src/main/java/io/vehiclehistory/api/OptionalProvider.java

```diff
@@ -0,0 +1,236 @@
+import io.vehiclehistory.api.model.Mileage;
+import io.vehiclehistory.api.model.Name;
+import io.vehiclehistory.api.model.Plate;
+import io.vehiclehistory.api.model.Policy;
+import io.vehiclehistory.api.model.PolicyStatus;
+import io.vehiclehistory.api.model.Production;
+import io.vehiclehistory.api.model.Registration;
+import io.vehiclehistory.api.model.RegistrationStatus;
+import io.vehiclehistory.api.model.Vehicle;
+import io.vehiclehistory.api.model.VehicleType;
+
+/**
+ * Created by m4lysh on 2015-04-06.
+ */
+public class OptionalProvider {
+
+    private Resources resources;
+
+    public OptionalProvider(Context context) {
+        this.resources = context.getResources();
+    }
+
+    public String getFirstVehicleRegistration(Vehicle vehicle) {
+        Registration registration = vehicle.getRegistration();
+        if (registration == null) {
+            return resources.getString(R.string.unknown_first_vehicle_registration);
+        }
+
+        String firstRegistration = registration.getFirstAt();
+
+        return firstRegistration != null
+                ? new DateFormatter().formatDateFromApi(firstRegistration)
+                : resources.getString(R.string.unknown_first_vehicle_registration);
+    }
+
+    public String getRegistrationStatus(Vehicle vehicle) {
+        Registration registration = vehicle.getRegistration();
+        if (registration == null) {
+            return resources.getString(R.string.unknown_registration_status);
+        }
+
+        RegistrationStatus status = registration.getStatus();
+
+        return status != null
+                ? status.toString()
+                : resources.getString(R.string.unknown_registration_status);
+    }
+
+    public String getMakePlusModel(Vehicle vehicle) {
+        Name name = vehicle.getName();
+        if (name == null) {
+            return resources.getString(R.string.unknown_car);
+        }
+
+        String model = name.getModel();
+        CarMake make = name.getMake();
+
+        if (model == null && make == null) {
+            return resources.getString(R.string.unknown_car);
+        }
+
+        StringBuilder sb = new StringBuilder();
+
+        if (make != null) {
+            sb.append(make.toString());
+            sb.append(" ");
+        }
+
+        if (model != null) {
+            sb.append(model);
+        }
+
+        return sb.toString();
+    }
+
+    public String getModel(Vehicle vehicle) {
+        Name name = vehicle.getName();
+        if (name == null) {
+            return resources.getString(R.string.unknown_model);
+        }
+
+        String model = name.getModel();
+
+        return model != null
+                ? model : resources.getString(R.string.unknown_model);
+    }
+
+    public String getMake(Vehicle vehicle) {
+        Name name = vehicle.getName();
+        if (name == null) {
+            return resources.getString(R.string.unknown_make);
+        }
+
+        CarMake make = name.getMake();
```

Comment:
If you mean correctly displaying new makes: That would require us to parse makes as text instead of enum. It would be best if api provided us with a value to display.